blockchain/chain_sync/src/sync.rs

```diff
@@ -230,12 +339,17 @@ where
     }
     /// Returns FullTipset from store if TipSetKeys exist in key-value store otherwise requests FullTipset
     /// from block sync
-    pub fn fetch_tipset(&self, _peer_id: PeerId, tsk: &TipSetKeys) -> Result<FullTipset, Error> {
+    async fn fetch_tipset(
+        &mut self,
+        peer_id: PeerId,
+        tsk: &TipSetKeys,
+        count: u64,
```

Comment:
isn't this always going to be 1? you are checking if a singular tipset exists in storage then requesting if not